app/components/traffic-measure/index.js

```diff
@@ -71,24 +74,64 @@ export default class TrafficMeasureIndexComponent extends Component {
       relations.map((relation) => relation.concept)
     );
 
+    yield this.fetchInstructions.perform();
+
     this.parseTemplate();
   }
 
+  @task
+  *fetchInstructions() {
+    //refresh instruction list from available signs
+    this.instructions = [];
+    this.inputTypes = ['text', 'number', 'date', 'location', 'codelist'];
+    for (let i = 0; i < this.signs.length; i++) {
+      const sign = this.signs[i];
+      const instructions = yield sign.templates;
+      for (let j = 0; j < instructions.length; j++) {
+        const instruction = instructions.objectAt(j);
+        this.instructions.push(instruction);
+      }
+    }
+    //remove input type instruction if there are none available and reset mappings with instructions
+    if (this.instructions.length != 0) {
+      this.inputTypes.push('instruction');
+    } else if (this.instructions.length == 0) {
+      if (this.inputTypes.indexOf('instruction') != -1) {
+        this.inputTypes.splice(this.inputTypes.indexOf('instruction'), 1);
+      }
+      this.mappings.forEach((e) => {
+        if (e.type == 'instruction') {
+          this.updateMappingType(e, 'text');
+        }
+      });
+    }
+    // eslint-disable-next-line
+    this.inputTypes = this.inputTypes;
```

Comment:
I don't think this a good solution. Consider:
 *  using a tracked array so you don't need this assignment (check with arne for the specific library)
 *  do the work on another array and then assign it
 *  use emberarray methods such as pushObject so changes are notified

Reply:
oki